### DISCUSSION TITLE
HyperHDR not detecting HyperSPI LED Device

### DISCUSSION THREAD

**Discussion by ericellb** [awawa-dev/HyperHDR#944] [General]
Hello,

I'll preface this with a couple things
- Raspberry Pi 3B+ running HyperHDR v20.0.0.0
- ESP32 with hyperspi_esp32_SK6812_RGBW_COLD.bin` firmware (Looks like this https://images.theengineeringprojects.com/image/main/2024/03/esp32-pinout.jpg)
- HyperSPI wiring between Raspberry PI / ESP32 is correct (Yes I have a ground between the Pi / ESP32)
- No smoothing
- Refresh time 0
- SPI Path /dev/spidev0.0
- Baudrate 2500000

In HyperHDR under LED Device, nothing is detected. When I disable / enable the LED Device, I see a `please wait....` warning that disapears after the countdown.

I then went into Remote Control and set the color to Purple for debugging.

I plugged the ESP32 into my PC and viewed the console via Putty, and I see logs similar to the following. I am assuming that I should see the RGB colors the same as the one I just set above. (Something that looks like purple, not 160,160,160 as we see below)

Changing the Color / Effects, shows the following. The frames seem okay, but the colors?

```
[02:26:44]HyperHDR frames: 60 (FPS), receiv.: 60, good: 60, incompl.: 0, mem1: 0, mem2: 0, heap: 310688
[02:26:44]RGBW => Gain: 255/255, red: 160, green: 160, blue: 160
[02:26:47]HyperHDR frames: 60 (FPS), receiv.: 60, good: 60, incompl.: 0, mem1: 0, mem2: 0, heap: 310688
[02:26:47]RGBW => Gain: 255/255, red: 160, green: 160, blue: 160
[02:26:50]HyperHDR frames: 60 (FPS), receiv.: 60, good: 60, incompl.: 0, mem1: 0, mem2: 0, heap: 310688
[02:26:50]RGBW => Gain: 255/255, red: 160, green: 160, blue: 160
```

I dont currently have an HDMI input plugged in, or the RGB LED strips connected. I am simply trying to view the data through the console and it seems like the correct colors are not being picked up.

**Comment by awawa-dev** [comment]
Hi

>(Something that looks like purple, not 160,160,160 as we see below)

this is calibration setting for AWA protocol (cold white), not the color that is being rendered.

> [02:26:44]HyperHDR frames: 60 (FPS), receiv.: 60, good: 60, incompl.: 0, mem1: 0, mem2: 0, heap: 310688

seems everything is OK

**Comment by ericellb** [comment]
Thank you! Everything seems to be working, must have just been a small ID10T error of working at 2AM :)
